src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ILScanner.cs

```diff
@@ -543,7 +543,12 @@ public ScannedDevirtualizationManager(NodeFactory factory, ImmutableArray<Depend
                             TypeDesc canonType = type.ConvertToCanonForm(CanonicalFormKind.Specific);
 
                             if (!canonType.IsDefType || !((MetadataType)canonType).IsAbstract)
```

Comment:
```suggestion
                            if (canonType is not MetadataType { IsAbstract: true })
```
Would this be an equivalent condition?

(We use this style earlier in this method.)